assembl/static2/js/app/utils/server.js

```diff
@@ -0,0 +1,16 @@
+/** *
+ * Use the standard assembl server location for API calls or Stargate
+ *
+ * @return String path - the path to use for the API call
+ * */
+export const basePath = () => {
+  const useStargate = document.getElementById('useStargate') ? document.getElementById('useStargate').value : 'false';
+
+  let path = `${window.location.protocol}//${window.location.host}`;
+
+  if (useStargate === 'true') {
+    path = `${window.location.protocol}//${window.location.hostname}:3000`; // move to config ?
```

Comment:
The port should be hard-coded? Pass as config as the comment suggests (pull from html).

Reply:
You're right, not too happy about pulling from HTML but it should not be hard-coded, will change

Reply:
Se should chat about this, not happy about people 'seeing' our port in the html code - it negates the fact it's hidden behind the react  - let's discuss

Reply:
Okay, alternatives:

1. Stargate API must have an API that returns the port, this API call made by Assembl and is awaited before any rendering (this must be the first API call of the application)
2. We pass a js variable in jinja2 which will contain our `configs`. We'll have to post-process our HTML prior to sending to client be minified/obfuscated (slows down time to first byte)
3. Do nothing on HTML. It's not the place of the html to be secure. It's the place of Stargate to be hardened and defend against attacks at this port

I like (3)

Reply:
Or:

4. Put a path under Assembl, so you benefit from TLS encryption and have nginx redirect requests to port 3000, wherever this machine is hosted.

Reply:
Can't do 4, makes me want to puke.
Let's do 3